analyzers/src/SonarAnalyzer.Common/Extensions/DiagnosticDescriptorExtensions.cs

```diff
@@ -18,10 +18,29 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using static SonarAnalyzer.Helpers.DiagnosticDescriptorFactory;
+
 namespace SonarAnalyzer.Extensions;
 
 public static class DiagnosticDescriptorExtensions
 {
+    public static bool HasMatchingScope(this DiagnosticDescriptor descriptor, Compilation compilation, bool isTestProject, bool isScannerRun)
```

Comment:
Can't we move this on reporting context?

Reply:
I've moved it to `SonarAnalysisContextBase` in #6563, that was even better place (after few other updates that will clean the air)